modules/evm/src/lib.rs

```diff
@@ -870,6 +886,15 @@ impl<T: Config> Pallet<T> {
 		Ok(())
 	}
 
+	/// Selfdestruct a contract at a given address.
+	///
+	/// - Ensure that the given maintainer is the contract maintainer.
```

Comment:
Try avoid just describe the code in English, should describe the parameters (and constraints), what happens (without all the details), and the returns

Also no need to have detailed docs for internal method unless the method is complicated.